conclave-host/src/main/kotlin/com/r3/conclave/host/internal/GramineEnclaveHandle.kt

```diff
@@ -30,12 +30,12 @@ class GramineEnclaveHandle(
         require(enclaveMode != EnclaveMode.MOCK)
         unzipIntoWorkingDir()
 
-        /**
-         * Not all threads in the enclave are necessarily the result of conclave calls.
-         * To minimise the likelihood of deadlocks, we don't allow conclave to use all available threads.
-         */
-        val maxConcurrentCalls = getEnclaveThreadCountFromManifest() / 2
-        check(maxConcurrentCalls > 0)
+        /** Load gramine enclave metadata. */
+        val metadataFile = workingDirectory / GRAMINE_ENCLAVE_METADATA
+        val maxConcurrentCalls = metadataFile.toFile().inputStream().use { inputStream ->
+            val metadataProperties = Properties().apply { load(inputStream) }
```

Comment:
Have a look at the docs for `Properties.reader`. Is it safe to use this considering we're writing the strings without any of the weird escaping that Properties does? We might need to simply roll our own "properties" reading code.